sdl_android/src/main/java/com/smartdevicelink/api/lockscreen/LockScreenManager.java

```diff
@@ -0,0 +1,274 @@
+
+		super(internalInterface);
+		this.context = new WeakReference<>(context);
+
+		// set initial class variables
+		hmiLevel = HMILevel.HMI_NONE;
+		driverDistStatus = false;
+
+		// setup the manager
+		readConfiguration(lockScreenConfig);
+		setupListeners();
+
+		// transition state
+		transitionToState(READY);
+	}
+
+	@Override
+	public void dispose(){
+		// send broadcast to close lock screen if open
+		context.get().sendBroadcast(new Intent(SDLLockScreenActivity.CLOSE_LOCK_SCREEN_ACTION));
+		// remove listeners
+		internalInterface.removeOnRPCNotificationListener(FunctionID.ON_HMI_STATUS, hmiListener);
+		internalInterface.removeOnRPCNotificationListener(FunctionID.ON_DRIVER_DISTRACTION, ddListener);
+		if (showOEMLogo) {
+			internalInterface.removeOnRPCNotificationListener(FunctionID.ON_SYSTEM_REQUEST, systemRequestListener);
+		}
+		lockScreenOEMIcon = null;
+		OEMIconUrl = null;
+
+		// transition state
+		transitionToState(SHUTDOWN);
+	}
+
+	////
+	// SETUP
+	////
+
+	/**
+	 * Reads configuration set in SDLManager Builder
+	 * @param lockScreenConfig - the configuration set.
+	 */
+	private void readConfiguration(LockScreenConfig lockScreenConfig){
+		lockScreenIcon = lockScreenConfig.getAppIcon();
+		lockScreenColor = lockScreenConfig.getBackgroundColor();
+		customView = lockScreenConfig.getCustomView();
+		lockScreenEnabled = lockScreenConfig.getEnabled();
+		showOEMLogo = lockScreenConfig.getShowOEMLogo();
+	}
+
+	/**
+	 * Adds 3 listeners that help determine whether or not a lockscreen should be shown.
+	 * This will change the variables that we hold in the manager to the newest values and then
+	 * usually call launchLockScreenActivity
+	 *
+	 * 1. ON_HMI_STATUS
+	 * 2. ON_DRIVER_DISTRACTION
+	 * 3. ON_SYSTEM_REQUEST (used for OEM Icon Downloading)
+	 */
+	private void setupListeners(){
+		// add hmi listener
+		hmiListener = new OnRPCNotificationListener() {
+			@Override
+			public void onNotified(RPCNotification notification) {
+				hmiLevel = ((OnHMIStatus)notification).getHmiLevel();
+				launchLockScreenActivity();
+			}
+		};
+		internalInterface.addOnRPCNotificationListener(FunctionID.ON_HMI_STATUS, hmiListener);
+
+		// set up driver distraction listener
+		ddListener = new OnRPCNotificationListener() {
+			@Override
+			public void onNotified(RPCNotification notification) {
+				// do something with the status
+				if (notification != null) {
+					OnDriverDistraction ddState = (OnDriverDistraction) notification;
+
+					if (ddState.getState() == DriverDistractionState.DD_ON){
+						Log.i(TAG, "DD On Notification");
+						// launch lock screen
+						driverDistStatus = true;
+						launchLockScreenActivity();
+					}else{
+						Log.i(TAG, "DD Off Notification");
+						// close lock screen
+						driverDistStatus = false;
+						context.get().sendBroadcast(new Intent(SDLLockScreenActivity.CLOSE_LOCK_SCREEN_ACTION));
```

Comment:
This is a potential NPE. The weak reference object will always hold a value but the object it references can be cleared.